Visualization/CombineBlastTables.R

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env Rscript
+
+# CombineBlastTables.R
+# Copyright Jackson M. Tsuji, 2018
+# Neufeld Research Group
+# See required R packages below.
+
+#####################################################
+## User variables: #################################
+RUN_COMMAND_LINE <- TRUE # If selected, all user input here is ignored, and terminal-based input is expected instead.
+
+# Set other user variables here
+if (RUN_COMMAND_LINE == FALSE) {
+  setwd("/Users/JTsuji/Downloads") # your working directory where files are stored
+  input_filenames <- c("subject1.csv", "subject2.csv", "subject3.csv") # NOTE: filename minux extesion will become the subject_name!
+  output_filename <- "combined_blast_tables.csv"
+}
+#####################################################
+
+#####################################################
+## Load required packages: ##########################
+library(getopt)
+library(tools)
+library(plyr)
+suppressMessages(library(dplyr))
+#####################################################
+
+# Hard-coded variables
+header_names <- c("qseqid", "sseqid", "pident", "evalue", "qcovhsp", "bitscore")
+
+
+# Function to assign command line input to variables, or throw help message
+parse_command_line_input <- function() {
+  
+  inputArgs <- commandArgs(trailingOnly = TRUE)
+  if (length(inputArgs) < 2) {
+    cat("CombineBlastTables.R: Combines BLAST tables and makes new column for sample ID based on the filenames.\n")
+    cat("Copyright Neufeld Lab, 2018\n")
+    cat("Contact Jackson M. Tsuji (jackson.@uwaterloo.ca) for bug reports or feature requests.\n\n")
+    
+    cat("Usage: CombineBlastTables.R subject1.csv subject2.csv ... subjectN.csv combined_blast_tables.csv\n\n")
+    
+    cat("Details:\n",
+        "subject1.csv, etc.: CSV BLAST tables for all individual samples (subjects for BLAST). Name should be the subject name [Required]\n",
+        "combined_blast_tables.csv: the output CSV file. [Required]\n\n")
+        
+    quit(status = 1)
+  }
+  
+  num_args <- length(inputArgs)
+  
+  # Make variables from provided input and save as global variables (<<-)
+  input_filenames <<- inputArgs[1:(num_args-1)]
+  output_filename <<- inputArgs[num_args]
+  
+}
+
+# Helper function for timestamps
+ts <- function() {
```

Comment:
The function name should be more explicit than **TS.** I shouldn't have to see ```ts()``` and then scroll all the way to the top of the page to find function definition and then read the function definition comment in order to figure out what the function does. This becomes especially confusing when you start importing function names from other files. Change it to something like 'generate_time_stamp()'.

Reply:
There also a third party logging package (https://stackoverflow.com/questions/1928332/is-there-any-standard-logging-package-for-r) but let's not worry about that for now.

Reply:
@LeeBergstrand Thanks for the suggestion -- I've never looked into R loggers before. Maybe I can try "futile.logger" in the future. For now, I've made the function name more explicit with the name 'timestamp'